app/src/main/java/com/cornellappdev/uplift/networking/FireStoreRepository.kt

```diff
@@ -0,0 +1,50 @@
+package com.cornellappdev.uplift.networking
+
+import android.util.Log
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.tasks.await
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class FireStoreRepository @Inject constructor(
+    private val fireStore: FirebaseFirestore
+) {
+
+    /**
+     * Checks if the specified email has been onboarded.
+     * @param email The email to check.
+     */
+    suspend fun getUserOnboarded(email: String, onError: () -> Unit, onSuccess: (Boolean) -> Unit) {
+        try {
+            val doc = fireStore.collection("user").document(email)
+                .get().await()
+
+            val user = doc.toObject(FirebaseUser::class.java)
+            onSuccess(user?.onboarded ?: false)
+        } catch (e: Exception) {
+            Log.e("FireStoreRepository", "Error getting user: ", e)
+            onError()
+        }
+    }
```

Comment:
Is uplift choosing to use firestore? There wasn't any particular need for resell to use firestore; I think we just decided to use that more or less. These functions' implementations hit firestore--we only need them if we are indeed using firestore for a database.